src/components/featured-article/macro.spec.js

```diff
@@ -0,0 +1,133 @@
+            });
+
+            test('THEN: the chart has the correct Id', () => {
+                expect($('[data-highcharts-base-chart]').attr('data-highcharts-id')).toBe('growth-chart');
+            });
+
+            test('THEN: the chart has the correct type', () => {
+                expect($('[data-highcharts-base-chart]').attr('data-highcharts-type')).toBe('line');
+            });
+        });
+
+        describe('WHEN: image is provided and chart is not', () => {
+            const $ = cheerio.load(renderComponent('featured-article', EXAMPLE_FEATURED_ARTICLE_WITH_IMAGE));
+
+            test('THEN: it passes accessibility checks', async () => {
+                const results = await axe($.html());
+                expect(results).toHaveNoViolations();
+            });
+
+            test('THEN: it displays the correct title text', () => {
+                expect($('.ons-featured__title').text()).toContain('Population Insights');
+            });
+
+            test('THEN: it links the title correctly', () => {
+                expect($('.ons-featured__title a').attr('href')).toBe('/people/population/insights');
+            });
+
+            test('THEN: it renders image with correct src', () => {
+                expect($('.ons-image__img').attr('src')).toBe('example.png');
+            });
+
+            test('THEN: it renders image with correct alt text', () => {
+                expect($('.ons-image__img').attr('alt')).toBe('Example alt text');
+            });
+        });
+    });
+
+    describe('GIVEN: Param: metadata', () => {
+        describe('WHEN: metadata with date and object is provided', () => {
+            const $ = cheerio.load(renderComponent('featured-article', EXAMPLE_FEATURED_ARTICLE_WITH_CHART));
+
+            test('THEN: it renders ISO date value correctly', () => {
+                expect($('time').attr('datetime')).toBe('2024-05-01');
+            });
+
+            test('THEN: it renders short date label correctly', () => {
+                expect($('time').text()).toBe('1 May 2024');
+            });
+
+            test('THEN: it displays metadata text', () => {
+                expect($('.ons-featured__item-attribute span').text()).toContain('Bulletin');
+            });
+        });
+
+        describe('WHEN: metadata object has no URL', () => {
+            const $ = cheerio.load(renderComponent('featured-article', EXAMPLE_FEATURED_ARTICLE_WITH_IMAGE));
+
+            test('THEN: it does not render metadata link', () => {
+                expect($('.ons-featured__attribute-link').length).toBe(0);
+            });
+
+            test('THEN: it still displays metadata text', () => {
+                expect($('.ons-featured__item-attribute span').text()).toContain('Article');
+            });
+        });
+
+        describe('WHEN: metadata is not provided', () => {
+            const articleWithoutMetadata = {
+                ...EXAMPLE_FEATURED_ARTICLE_WITH_CHART,
+                metadata: undefined,
```

Comment:
Rather than set it to undefined here we could have this example have no metadata and then add it in the tests that need metadata set. This would work with my comment about passing in metadata for the required params tests